### PR TITLE
7_POST_/api/articles/article_id/comments

### PR DESCRIPTION
**Pull request by TomMetcalf**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -156,45 +156,115 @@ describe('/api/articles', () => {
...
+        });
+        test('POST - status: 400 - username is not valid', () => {
+            const testComment = {
+                username: 'jon123',
+                body: 'this is the best news ever!',
+            };
+            return request(app)
+                .post('/api/articles/6/comments')
+                .send(testComment)
+                .expect(400)
```
Comment (pippaamy) [inline __tests__/app.test.js:261]: would this be a 400 error? 👀

File: __tests__/app.test.js [hunk 2]
```diff
@@ -156,45 +156,115 @@ describe('/api/articles', () => {
...
+                    const postedComment = response.body.comment;
+                    expect(postedComment.comment_id).toBe(19);
+                    expect(postedComment.body).toBe(
+                        'this is the best news ever!'
+                    );
+                    expect(postedComment.author).toBe('icellusedkars');
+                    expect(postedComment.votes).toBe(0);
+                    expect(typeof postedComment.created_at).toBe('string');
+                });
+        });
```
Comment (pippaamy) [inline __tests__/app.test.js:224]: can we add another error test that there are missing fields required to post

File: __tests__/app.test.js [hunk 3]
```diff
@@ -156,45 +156,115 @@ describe('/api/articles', () => {
...
     });
-    test('GET - status: 400 - requested id is not valid', () => {
-        return request(app)
-            .get('/api/articles/nonsense/comments')
-            .expect(400)
-            .then((response) => {
-                expect(response.body).toEqual({ msg: 'Bad Request' });
-            });
+    describe('POST', () => {
+        test('POST - status: 201 - responds with newly created comment', () => {
```
Comment (pippaamy) [inline __tests__/app.test.js:205]: great! let's add another 201 test that it ignores unnecessary properties!

File: controllers/comment.controllers.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const { createComment } = require('../models/comment.models');
+
+exports.postComment = (req, res, next) => {
+    const { article_id } = req.params;
+    const { username, body } = req.body;
+    createComment({ article_id, username, body })
+        .then((comment) => {
+            res.status(201).send({ comment });
```
Comment (pippaamy) [inline controllers/comment.controllers.js:8]: fab!

File: models/comment.models.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const connection = require('../db/connection');
+
+exports.createComment = (newComment) => {
+    const { article_id, username, body } = newComment;
+    return connection
+        .query('SELECT * FROM users WHERE username = $1', [username])
+        .then((result) => {
+            if (result.rows.length === 0) {
+                return Promise.reject({
```
Comment (pippaamy) [inline models/comment.models.js:9]: we shouldn't be sending the error code back here

**Review by pippaamy — COMMENTED** [review]
good job, lets make those changes and pop in another pr :)

**Review by robert112233 — APPROVED** [review]
Merge, great work !

**Pull request merged**